src/ProjectNeon/Assets/Features/Battle/Cards/CardResolutionZone.cs

```diff
@@ -9,6 +9,7 @@ public class CardResolutionZone : ScriptableObject
     [SerializeField] private CardPlayZone physicalZone;
     [SerializeField] private CardPlayZone playedDiscardZone;
     [SerializeField] private GameEvent onFinished;
+    [SerializeField] private BattleState battleState;
```

Comment:
This is a circular reference. You can not depend on something which depends on you.